mac/KeymanEngine4Mac/KeymanEngine4MacTests/PlatformTest.kmn

```diff
@@ -0,0 +1,80 @@
+﻿store(&TARGETS) 'any'
+store(&NAME) 'PlatformTest'
+begin Unicode > use(main)
+
+group(main) using keys 
+
+platform('touch') + [K_P] > ' Touch'
+platform('hardware') + [K_P] > ' Hardware'
++ [K_P] > ' [UI Undefined]'
+
+platform('TOUCH') + [SHIFT K_P] > ' Touch'
+platform('HARDWARE') + [SHIFT K_P] > ' Hardware'
++ [SHIFT K_P] > ' [UI Undefined]'
+
+platform('windows') + [K_O] > ' Windows'
+platform('android') + [K_O] > ' Android'
+platform('iOS') + [K_O] > ' iOS'
+platform('linux') + [K_O] > ' Linux'
+platform('macosx') + [K_O] > ' macOS'
++ [K_O] > ' [OS Undefined]'
+
+platform('WINDOWS') + [SHIFT K_O] > ' Windows'
+platform('ANDROID') + [SHIFT K_O] > ' Android'
+platform('IOS') + [SHIFT K_O] > ' iOS'
+platform('LINUX') + [SHIFT K_O] > ' Linux'
+platform('MacOSX') + [SHIFT K_O] > ' macOS'
++ [SHIFT K_O] > ' [OS Undefined]'
+
+platform('mac') + [K_L] > ' mac'
++ [K_L] > ' [Not a mac]'
+platform('MAC') + [SHIFT K_L] > ' mac'
++ [SHIFT K_L] > ' [Not a mac]'
+platform('macos') + [K_K] > ' macOS'
++ [K_K] > ' [Not macos]'
+platform('MacOS') + [SHIFT K_K] > ' macOS'
++ [SHIFT K_K] > ' [Not macos]'
+
+platform('tablet') + [K_I] > ' Tablet'
+platform('phone') + [K_I] > ' Phone'
+platform('desktop') + [K_I] > ' Desktop'
++ [K_I] > ' [FF Undefined]'
+
+platform('TABLET') + [SHIFT K_I] > ' Tablet'
+platform('Phone') + [SHIFT K_I] > ' Phone'
+platform('DeskTop') + [SHIFT K_I] > ' Desktop'
++ [SHIFT K_I] > ' [FF Undefined]'
+
+platform('web') + [K_U] > ' Web'
+platform('native') + [K_U] > ' Native'
++ [K_U] > ' [Nativeness Undefined]'
+
+platform('Web') + [SHIFT K_U] > ' Web'
+platform('NATIVE') + [SHIFT K_U] > ' Native'
++ [SHIFT K_U] > ' [Nativeness Undefined]'
+
+platform('ie') + [K_Y] > ' IE'
+platform('chrome') + [K_Y] > ' Chrome'
+platform('firefox') + [K_Y] > ' Firefox'
+platform('safari') + [K_Y] > ' Safari'
+platform('opera') + [K_Y] > ' Opera'
++ [K_Y] > ' [Browser Undefined]'
+
+platform('IE') + [SHIFT K_Y] > ' IE'
+platform('Chrome') + [SHIFT K_Y] > ' Chrome'
+platform('FireFox') + [SHIFT K_Y] > ' Firefox'
+platform('Safari') + [SHIFT K_Y] > ' Safari'
+platform('OPERA') + [SHIFT K_Y] > ' Opera'
++ [SHIFT K_Y] > ' [Browser Undefined]'
+
+platform('tablet') + [K_T] > ' Mobile'
+platform('phone') + [K_T] > ' Mobile'
++ [K_T] > ' [Mobile Undefined]'
+
+platform('tablet ie web') + [K_M] > ' tablet ie web'
+platform('touch phone safari web') + [K_M] > ' phone safari web'
+platform('Windows desktop native hardware') + [K_M] > ' Windows desktop native hardware'
+platform('Linux desktop native') + [K_M] > ' Linux desktop native'
+platform('macOS native desktop hardware Safari') + [K_M] > ' macOS native hardware Safari'
+platform('macOS native desktop hardware') + [K_M] > ' macOS native desktop hardware'
++ [K_M] > ' [No match]'

```

Comment:
For maintainability and consistency, it would be nice if this was identical to `windows/src/test/manual-tests/platform-rules/platformtest.kmn`.

For example, 
```
platform('edge')    > ' edge'
``` 
was added for #969

Reply:
i.e. consolidated into one .kmn for all platforms. It would give consistent results for the giant spreadsheet too.

Reply:
Fair point. And as an added bonus, a single test would accomplish most of what is needed (except for the "macos" and "mac" aliases for "macosx" and the case-insensitive testing). One downside of that is that by reordering the rules in the one for Mac, I was able to ensure that all the in correct ones do return false. I don't think we can accomplish that on all the different platforms if we use a single keyboard, unless we create a lot more groups with different orderings and casings (and use a/A,b/B,c/C,d/D,e/E to access them). I guess it doesn't have to be small or pretty, but it would make it a little less maintainable. An alternative would be to assign a different key to each platform token to allow more fine-grained testing, but then the number of tests required would go way up! For unit testing, I did consider just exposing the underlying method so I could write more understandable tests, but I was a little reticent to do that.

Reply:
I'm happy if we just run the windows/src/test/manual-tests/platform-rules/platformtest.kmn and get that to pass. Attacking these tests from different directions is good; verifies the engine a lot better.